src/Domain/RuleInterface.php

```diff
@@ -23,4 +23,8 @@ public function getRequirements(): array;
     public function getType(): string;
 
     public function getDescription(): ?string;
+
+    public function matchesNode(NodeInterface $node): bool;
```

Comment:
I would call it simply matches, because when you have the implementation, it's like a self explanatory text:

    $rule->matches($node)

Reply:
(and some for the other method, personally I would remove the inNodes part)